src/decoder/mod.rs

```diff
@@ -494,56 +927,110 @@ where
+    ///
+    /// The upper bound is always `None` since the decoder loops indefinitely.
+    /// This differs from non-looped decoders which may provide a finite upper bound.
+    ///
+    /// Note that even with an active decoder, reaching the end of the stream may result
+    /// in the decoder becoming inactive if seeking back to the start fails.
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
-        self.0.size_hint()
+        (
+            self.inner.as_ref().map_or(0, |inner| inner.size_hint().0),
+            None,
+        )
     }
 }
 
 impl<R> Source for LoopedDecoder<R>
 where
     R: Read + Seek,
 {
+    /// Returns the current span length of the underlying decoder.
+    ///
+    /// Returns `None` if there is no active decoder.
     #[inline]
     fn current_span_len(&self) -> Option<usize> {
-        self.0.current_span_len()
+        self.inner.as_ref()?.current_span_len()
     }
 
+    /// Returns the number of channels in the audio stream.
+    ///
+    /// Returns the default channel count if there is no active decoder.
     #[inline]
     fn channels(&self) -> ChannelCount {
-        self.0.channels()
+        self.inner
+            .as_ref()
+            .map_or(ChannelCount::default(), |inner| inner.channels())
     }
 
+    /// Returns the sample rate of the audio stream.
+    ///
+    /// Returns the default sample rate if there is no active decoder.
     #[inline]
     fn sample_rate(&self) -> SampleRate {
-        self.0.sample_rate()
+        self.inner
+            .as_ref()
+            .map_or(SampleRate::default(), |inner| inner.sample_rate())
     }
 
+    /// Returns the total duration of this audio source.
+    ///
+    /// Always returns `None` for looped decoders since they have no fixed end point -
+    /// they will continue playing indefinitely by seeking back to the start when reaching
+    /// the end of the audio data.
     #[inline]
     fn total_duration(&self) -> Option<Duration> {
         None
     }
 
+    /// Attempts to seek to a specific position in the audio stream.
+    ///
+    /// # Errors
+    ///
+    /// Returns `SeekError::NotSupported` if:
+    /// - There is no active decoder
+    /// - The underlying decoder does not support seeking
+    ///
+    /// May also return other `SeekError` variants if the underlying decoder's seek operation fails.
+    ///
+    /// # Note
+    ///
+    /// Even for looped playback, seeking past the end of the stream will not automatically
+    /// wrap around to the beginning - it will return an error just like a normal decoder.
+    /// Looping only occurs when reaching the end through normal playback.
     fn try_seek(&mut self, pos: Duration) -> Result<(), SeekError> {
-        self.0.try_seek(pos)
+        match &mut self.inner {
+            Some(inner) => inner.try_seek(pos),
+            None => Err(SeekError::NotSupported {
+                underlying_source: "No decoder available",
```

Comment:
this can only be None if the "reset" seek failed. 

Maybe we can do 
```rust
struct LoopBackFailure;

impl Error for LoopBackFailure {
  .. some msg like "Looped source ended when it failed to loop back"
}

SeekError::Other(Box::new(LoopBackFailure))
```

Reply:
I now re-used the existing `DecoderError::IoError`. Seems good enough without introducing another error type, do you agree?

Reply:
for context this is with IoError
```rust
            None => Err(SeekError::Other(Box::new(DecoderError::IoError(
                "Looped source ended when it failed to loop back".to_string(),
            )))),
```

and that looks great to me :+1: